routers/user/getOneStudent.ts

```diff
@@ -0,0 +1,45 @@
+import { NextFunction, Request, Response } from "express";
+import { Roles } from "../../src/entities/types/Roles";
+import { ValidationError } from "../../utils/errorsHandler";
+import { StudentsRating } from "../../src/entities/studentsRating/studentsRating.entity";
+import { myDataSource } from "../../config/database.configuration";
+import { StudentStatus } from "../../src/entities/types/studentsData";
+import { UserPayloadData } from "../../utils/createTokens";
+import { DestructuringToGetOne } from "../../utils/destructuringGetOne";
+
+type RequestAndPayloadUser = Request & UserPayloadData;
+
+export const getOneStudent = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const { id, role } = req.user as RequestAndPayloadUser;
+
+    if (role !== Roles.HR && role !== Roles.STUDENT) throw new ValidationError('Access denied.', 401);
+
+    const userId = req.params.id;
+
+    const results = await myDataSource
```

Comment:
I added relation to StudentsRating, so we can load relatiron easier just by myDataSource.getRepository(User).findOneOrFail({ where: { userId }, relations: [studentsRating]}). But We can make it simpler in the future

Reply:
Ok. nie znałem tego, chętnie się doszkolę.